executor/aggfuncs/func_count.go

```diff
@@ -280,54 +280,61 @@ func (e *countOriginalWithDistinct) UpdatePartialResult(sctx sessionctx.Context,
 func (e *countOriginalWithDistinct) evalAndEncode(
 	sctx sessionctx.Context, arg expression.Expression,
 	row chunk.Row, buf, encodedBytes []byte,
-) ([]byte, bool, error) {
+) (_ []byte, isNull bool, err error) {
 	switch tp := arg.GetType().EvalType(); tp {
 	case types.ETInt:
-		val, isNull, err := arg.EvalInt(sctx, row)
+		var val int64
+		val, isNull, err = arg.EvalInt(sctx, row)
 		if err != nil || isNull {
-			return encodedBytes, isNull, errors.Trace(err)
+			break
 		}
 		encodedBytes = appendInt64(encodedBytes, buf, val)
```

Comment:
```suggestion
		val, isNull, err = arg.EvalInt(sctx, row)
		if err == nil && !isNull {
			encodedBytes = appendInt64(encodedBytes, buf, val)	
		}

Reply:
It's ok here.
I'll change this format in master.